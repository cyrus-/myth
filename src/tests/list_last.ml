let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

type natopt =
  | None
  | Some of nat

let list_last : list -> natopt |>
  { [] => None
  | [1] => Some (1)
  | [2] => Some (2)
  | [2; 1] => Some (1)
  | [1; 2] => Some (2)
  | [3; 2; 1] => Some (1)
  } = ?"
