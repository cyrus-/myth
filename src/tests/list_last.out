(* Generated Thu Nov 13 00:35:27 2014 *)
let list_last : list -> natopt =
  let rec f1 (l1:list) : natopt =
    match l1 with
      | Nil -> None
      | Cons (n1, l2) -> (match l2 with
                            | Nil -> Some (n1)
                            | Cons (n2, l3) -> f1 l2)
  in
    f1
;;
